cli/internal/oauth/manager/util.go

```diff
@@ -0,0 +1,43 @@
+package manager
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/cli/cli/config/credentials"
+	"github.com/docker/cli/cli/version"
+	"github.com/shirou/gopsutil/v3/host"
+)
+
+const (
+	audience = "https://hub.docker.com"
+	tenant   = "login.docker.com"
+	clientID = "DHWuMefQ1v4lxENpz8oUYH50yYSwyPvi"
+)
+
+func NewManager(store credentials.Store) (*OAuthManager, error) {
+	hostinfo, err := host.Info()
```

Comment:
I was, erm, curious where all the extra dependencies originated from :see_no_evil:;

- `github.com/tklauser/go-sysconf`
- `github.com/tklauser/go-sysconf` (number of online CPUs?)
- `github.com/shoenig/go-m1cpu` (CPU info for m1 CPUs)
- `github.com/power-devops/perfstat` (handling of `libperfstat` ??
- `github.com/go-ole/go-ole` (loading of Windows OLE objects)
- `github.com/yusufpapurcu/wmi` (WQL interface to Windows Management Instrumentation (WMI))


And those all look to be related to this single `host.Info()`; `gopsutil/host.Info()` is doing a _lot_ of things; https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host.go#L65-L117

- Getting information about the distro-version, which;
   - on Linux involves trying to parse `/etc/os-release` and more; https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host_linux.go#L173-L321
   - on macOS involves shelling out to `sw_vers -productVersion` https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host_darwin.go#L98-L124
   - and on Windows invoking the infamous tricks to _try_ to get a version, because _there is no official API for this_ https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host_windows.go#L146-L234
   - ^^ on Windows that's invoked _twice_ because it uses PlatformVersion as KernelVersion
- Getting the kernel architecture, which on Linux/Darwin is just `unix.Utsname`, but on Windows involves parsing processor architectures; https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/host/host_windows.go#L251-L280
- Trying to detect Virtualization, which is only implemented on Linux, but there it's a _lot_, including (again) parsing `/etc/os-release`; enough for the library author to implement caching, but that won't do much for a short-lived process like the CLI; https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/internal/common/common_linux.go#L168-L346
- It also tries to get `BootTime`, and `UpTime`, which again do a lot, including shelling out to various binaries, and it even has special cases for `docker`; caching was implemented there as well, but won't help us here; https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/internal/common/common_linux.go#L63-L119
- It tries to obtain the number of running processes on the machine, which on Linux means listing all files in `/proc`, and for each of them trying to see if it can convert them to an Integer (in which case it's a process-id); https://github.com/shirou/gopsutil/blob/865b8c3f58d6ee4c47dab074fbf3c4b037abfd96/internal/common/common_linux.go#L41-L61
    - :point_up: which makes me want to sit in a corner and cry
- It tries to obtain a `HostID` (`sys/kernel/random/boot_id`, on Linux, but shelling out to `ioreg -rd1 -c IOPlatformExpertDevice` on Darwin, and using the registry on Windows)